homeassistant/components/switch/pilight.py

```diff
@@ -26,15 +25,9 @@
 
 DEPENDENCIES = ['pilight']
 
+# Command scheme is arbitrary JSON with at least one entry
 COMMAND_SCHEMA = vol.Schema({
-    vol.Optional(CONF_PROTOCOL): cv.string,
-    vol.Optional('on'): cv.positive_int,
-    vol.Optional('off'): cv.positive_int,
-    vol.Optional(CONF_UNIT): cv.positive_int,
-    vol.Optional(CONF_UNITCODE): cv.positive_int,
-    vol.Optional(CONF_ID): cv.positive_int,
-    vol.Optional(CONF_STATE): cv.string,
-    vol.Optional(CONF_SYSTEMCODE): cv.positive_int,
+    vol.Required(vol.Any(str)): vol.Any(str, int, float)
```

Comment:
Probably want to use `vol.Coerce(str)` for key, that way you convert it to a string if it is not.